src/services/rfqm_service.ts

```diff
@@ -439,6 +450,118 @@ export class RfqmService {
         return Math.random() > 0.5 ? Promise.reject('failing') : Promise.resolve();
     }
 
+    /**
+     * Submit a transaction, re-submit if necessary, and poll for resolution
+     */
+    public async completeSubmissionLifecycleAsync(
+        orderHash: string,
+        workerAddress: string,
+        callData: string,
+    ): Promise<void> {
+        const submissions: { [transactionHash: string]: RfqmTransactionSubmissionEntity } = {};
+
+        let gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
+
+        const firstSubmission = await this._submitTransactionAsync(orderHash, workerAddress, callData, gasPrice);
+        submissions[firstSubmission.transactionHash!] = firstSubmission;
+
+        let isFinishedMonitoring = false;
+        let shouldTryResubmitting = true;
```

Comment:
hmm... having two booleans strikes me as a little complex. Is it possible to get away with just one? I'll noodle on this a bit too

Reply:
ok, after thinking about it a little, I think I see that this is loop is a two phased process:
- submitting (and resubmitting if the gas prices moves against our highest gas price so far)
- waiting for finality


I wonder if we could simplify the code by breaking these into two distinct processes. Not sure if possible, but striving for code that at a high level looks like:
 ```
await mineTransaction(...);
await achieveTransactionFinality(...);
```

Reply:
I chunked up the logic a bit more to make this a bit more clear, tho not quite as suggested